lib/utils.js

```diff
@@ -62,16 +62,30 @@ export function getDeviceType(deviceType) {
  * @returns {Promise<String>} the most recent compatible version.
  */
 export async function resolveVersion(deviceType, versionOrRange) {
-	const version = await balena.models.os.getMaxSatisfyingVersion(
-		deviceType,
-		versionOrRange,
-	);
+	const version = isESR(versionOrRange)
+		? versionOrRange
+		: await balena.models.os.getMaxSatisfyingVersion(
+				deviceType,
+				versionOrRange,
+		  );
```

Comment:
@thgreasi, in my testing, this hacky approach allows `balena os download` to succeed with ESR versions, although losing the ability of specifying a semver range like `>2021.01.1 <2021.04.1`, which I would then document in `balena help os download`.

Reply:
@pdcastro I guess we wouldn't even need to do this check, if the SDK's `os.getMaxSatisfyingVersion()` was switched to use the hostApp releases, so that it also includes ESR releases right?

Reply:
I also can't find any other place in the CLI/SDK that getMaxSatisfyingVersion is used, so I'm guessing it's only used for OS downloads, which we do not support in oB anyway.

Reply:
We would not need to do this check if the SDK's `os.getMaxSatisfyingVersion()` understood ESR versions.

Reply:
I guess it would be better to keep this for the next major, since I can imagine the case that a user w/o access to ESR is using `>2.11.0` as a range and ends up getting an error b/c this now matches an ESR version (which the user doesn't have access to) causing the method to now throw.